lib/Models/Terria.ts

```diff
@@ -2072,8 +2072,10 @@ async function interpretStartData(
   errorSeverity?: TerriaErrorSeverity,
   showConversionWarning = true
 ) {
-  const containsStory = (initSource: InitSourceData) =>
-    Array.isArray(initSource.stories) && initSource.stories.length;
+  const containsStory = (initSource: InitSourceData | string) =>
```

Comment:
Why does this need to accept strings too but then we check if its json? Just for my learning. Is this the best place to have this union type or could we define InitSourceData as a union with string? Whats happening here?

Reply:
I have removed the function as it was a bit weird.

It is checking init sources from here

https://github.com/TerriaJS/terriajs/blob/bef8fb2ecea8a3f14ca7be28c1e0da0c1e302a12/lib/Models/InitSource.ts#L43-L47

which supports both string and `InitSourceData`

I have added some comments to make it more clear